### DISCUSSION TITLE
x-show removes other display values from x-bind or inline styles

### DISCUSSION THREAD

**Discussion by ekwoka** [alpinejs/alpine#3502] [7. Bugs (archived)]
Is this intentional?

```html
<div x-data x-show="true" :style="'display: flex'" style="display: flex;"></div>
```

it will just remove display entirely.

Is this intentional for a reason?

I feel like the expected behavior would actually be that when the thing is shows, it's original display attribute is returned to it. (most of us just use css classes to define the initial, but this seems like it matters)

**Comment by danddanddand** [comment]
It's for sure coded like that. 

https://github.com/alpinejs/alpine/blob/6142a83956685b4ddf67d95cf70a46fa3efe5fdf/packages/alpinejs/src/directives/x-show.js#L22

I've never had issue with it since I use Tailwind. Work arounds would be trivial if this was real problem for them. You could make PR to cache old value I guess.
